tests/app/main/views/test_search.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from lxml import html
+
+from ...helpers import LoggedInApplicationTest
+
+
+class TestSearchView(LoggedInApplicationTest):
+    @pytest.mark.parametrize("role,expected_code", [
+        ("admin", 200),
+        ("admin-ccs-category", 200),
+        ("admin-ccs-sourcing", 200),
+        ("admin-manager", 403),
+        ("admin-framework-manager", 200),
+    ])
+    def test_find_suppliers_and_services_page_is_only_accessible_to_specific_user_roles(
+        self, role, expected_code
+    ):
+        self.user_role = role
+        response = self.client.get('/admin/find-suppliers-and-services')
+        actual_code = response.status_code
+        assert actual_code == expected_code, "Unexpected response {} for role {}".format(actual_code, role)
+
+    @pytest.mark.parametrize("role, expected_header", [
+        ("admin", "Edit supplier accounts or view services"),
+        ("admin-ccs-category", "Edit suppliers and services"),
+        ("admin-ccs-sourcing", "Edit supplier declarations"),
+        ("admin-framework-manager", "View suppliers and services"),
+    ])
+    def test_should_show_find_suppliers_and_services_page_with_role_appropriate_header(self, role, expected_header):
+        self.user_role = role
+        response = self.client.get('/admin/find-suppliers-and-services')
+        page_html = response.get_data(as_text=True)
+        document = html.fromstring(page_html)
+        header = document.xpath(
+            '//header[@class="page-heading page-heading-without-breadcrumb"]//h1/text()')[0].strip()
+
+        assert response.status_code == 200
+        assert header == expected_header
+
+    @pytest.mark.parametrize("role,form_1_exists,form_2_exists,form_3_exists", [
```

Comment:
It'd be good to have these as `supplier_name_form`, `supplier_duns_form`, `service_id_form` rather than form1, form2 etc.